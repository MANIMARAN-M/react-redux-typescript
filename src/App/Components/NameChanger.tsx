import React, { useState, FC, FormEvent } from "react";
import { Link } from "react-router-dom";

interface NameDetails {
  name: string;
  onName: any;
}

const NameChanger: FC<{}> = (): JSX.Element => {
  const [Names, setNames] = useState<NameDetails>({
    onName: "",
    name: "",
  });
  const InputHandle = (e: FormEvent<HTMLInputElement>): void => {
    setNames({ ...Names, onName: e.currentTarget.value });
  };
  const NameHandle = (): void => {
    setNames({ ...Names, name: Names.onName });
  };

  console.log("render 2");
  return (
    <div className="HomeInputSection" data-testid="inputtest-1">
      <div>
        <h1 data-testid="inputtest-2">Enter somthing here</h1>
        <input
          type="text"
          value={Names.onName}
          onChange={(e) => InputHandle(e)}
        />
        <button onClick={NameHandle}>Enter</button>
        <h4>Result:</h4>
        <h3>{Names.name}</h3>
        <Link to="/listuser">
          <button className="LinkBtnUser">Go to User Page</button>
        </Link>
        <Link to="/reduxfetch">
          <button className="LinkBtnUser">Go to Redux Page</button>
        </Link>
      </div>
    </div>
  );
};

export default NameChanger;
