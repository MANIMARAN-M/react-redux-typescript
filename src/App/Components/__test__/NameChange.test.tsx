import NameChanger from "../NameChanger";
import { shallow, configure } from "enzyme";
import "@testing-library/jest-dom/extend-expect";
import { cleanup } from "@testing-library/react";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });
afterEach(cleanup);
describe("Namechanger component tests", () => {
  const component = shallow(<NameChanger />);
  it("test sample checking", () => {
    expect(2 + 2).toBe(4);
  });
  test("It should mount", () => {
    expect(component.length).toBe(1);
  });
  it("should input length check", () => {
    const input = component.find("input");
    expect(input).toHaveLength(1);
  });
  it("should button length and content check", () => {
    const button = component.find("button");
    expect(button).toHaveLength(3);
    // expect(component.find("button").text()).toEqual("Enter");
  });
  it("should render our Snapshots correctly", () => {
    expect(component).toMatchSnapshot();
  });
});
