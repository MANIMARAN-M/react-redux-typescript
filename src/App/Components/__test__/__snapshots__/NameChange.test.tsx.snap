// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Namechanger component tests should render our Snapshots correctly 1`] = `ShallowWrapper {}`;
